src/installer/installer.go

```diff
@@ -167,10 +168,25 @@ func (i *installer) InstallNode() error {
 	}
 	//update installation progress
 	i.UpdateHostInstallProgress(models.HostStageRebooting, "")
-	//reboot
-	if err = i.ops.Reboot(); err != nil {
+	// if ironic-agent exists on the host stop the assisted-agent service and exit, else reboot
```

Comment:
what's happening is pretty obvious from the code, could be nice instead to have a comment explaining _why_ we'd want to do that